blog: post urbit precepts and precepts discussion

urcades opened the pull request:
Posting the split version of Precepts and its following Discussion

urcades committed:
blog: post urbit precepts and precepts discussion

Posting the split version of Precepts and its following Discussion

urcades commented:
Hoping some of these AZ deep cut imageries are appreciated – Sedona is a wonderful town

matildepark commented on content/blog/precepts-discussion.md:
```diff
@@ -0,0 +1,769 @@
+description = "Urbit is a principled approach to system design and programming, but it's often not obvious what those technical principles are.  This is a list of the technical maxims that define Urbit's approach to engineering."
+[extra]
+author = "Philip Monk"
+ship = "~wicdev-wisret"
```
`~wicdev-wisryt`, pronounced "wicdev was right"

urcades replied:
lmfao, wonderful, ty

matildepark commented on content/blog/precepts-discussion.md:
```diff
@@ -0,0 +1,769 @@
++++
+title = "Precepts: Discussion"
+date = 2020-03-14
+description = "Urbit is a principled approach to system design and programming, but it's often not obvious what those technical principles are.  This is a list of the technical maxims that define Urbit's approach to engineering."
```
try this, adjusted from the opener as a description for the discussion: `The precepts aren’t arguments. We discuss and justify them here.`

matildepark commented on content/blog/precepts-discussion.md:
```diff
@@ -0,0 +1,769 @@
+Dually, a perfect testing suite can guarantee that `S` is a subset of `T` — no correct program fails the tests. This is vacuously true if you have no tests, for then `T = U`. So you want `T` to be as small as possible while still being a superset of `S` — you don't want to reject valid programs if you can help it. But you can't make `T == S` — there will always be some programs that are invalid b ...
+
+(It is possible to create a static type system that has `S` subset `T` property described above for a testing suite — see Erlang's Dialyzerand (related
+work)[http://user.it.uu.se/~tobiasl/publications/contracts.pdf].)
```
should be `[related work](http://user.it.uu.se/~tobiasl/publications/contracts.pdf)`

matildepark commented on content/blog/precepts-discussion.md:
```diff
@@ -0,0 +1,769 @@
+joke because off-by-one errors are not hard at all. They're mostly a matter of discipline, and there's a couple of conventions that help. If you're looping on a zero-indexed variable, use `i < num_loops`. If you're looping on a one-indexed variable, use `i <= num_loops`.
+
+However, cache invalidation and naming things are genuinely hard because they must be solved ad hoc. There is no one-size-fits-all cache invalidation strategy, and possibly there never will be. Similar for naming. So you must actually comprehend and work within each individual
+context.
```
Does the site deal well with the newline here? idk

matildepark replied:
It does, so resolved ~

matildepark commented on content/blog/precepts.md:
```diff
@@ -0,0 +1,355 @@
++++
+title = "Precepts"
+date = 2020-03-14
+description = "Urbit is a principled approach to system design and programming, but it's often not obvious what those technical principles are.  This is a list of the technical maxims that define Urbit's approach to engineering."
```
Simplify for the Twitter cards, lol: `The technical maxims that define Urbit's approach to engineering.`

urcades replied:
good call

matildepark commented on content/blog/precepts-discussion.md:
```diff
@@ -0,0 +1,769 @@
++++
+title = "Precepts: Discussion"
+date = 2020-03-14
```
These both should be adjusted for today's date.

matildepark commented on content/blog/precepts.md:
```diff
@@ -0,0 +1,355 @@
+description = "Urbit is a principled approach to system design and programming, but it's often not obvious what those technical principles are.  This is a list of the technical maxims that define Urbit's approach to engineering."
+[extra]
+author = "Philip Monk"
+ship = "~wicdev-wisret"
```
See previous

urcades replied:
fixed!

matildepark reviewed (changes requested):
Gonna look at this on the site now ...

matildepark commented:
Blockquotes are unstyled; just wondering if you have an opinion on that ... I prefer it to excessive margins, but maybe we want to italicise them or provide some sort of style hint that it's a quote?

![image](https://user-images.githubusercontent.com/20846414/76981639-41f29980-6911-11ea-8562-f0e6f4b0a324.png)

urcades commented:
> Blockquotes are unstyled; just wondering if you have an opinion on that ... I prefer it to excessive margins, but maybe we want to italicise them or provide some sort of style hint that it's a quote?
> 
> ![image](https://user-images.githubusercontent.com/20846414/76981639-41f29980-6911-11ea-8562-f0e6f4b0a324.png)

This was how we had previous specified quotes to look:

<img width="774" alt="Screen Shot 2020-03-18 at 1 00 22 PM" src="https://user-images.githubusercontent.com/1195363/76986837-98170b00-6918-11ea-9938-c3adcd52fec7.png">

Is this something that'd be implemented via main.scss?

matildepark commented:
Yeah that’s easy. border-left, padding-left, font-style: italic, on I think body.essay’s blockquotes — probably already in scss

—
~haddef-sigwen
https://urbit.org

> On Mar 18, 2020, at 1:02 PM, édouard <notifications@github.com> wrote:
> 
> 
> Blockquotes are unstyled; just wondering if you have an opinion on that ... I prefer it to excessive margins, but maybe we want to italicise them or provide some sort of style hint that it's a quote?
> 
> 
> This was how we had previous specified quotes to look:
> 
> 
> Is this something that'd be implemented via main.scss?
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

urcades commented:
> Yeah that’s easy. border-left, padding-left, font-style: italic, on I think body.essay’s blockquotes — probably already in scss — ~haddef-sigwen https://urbit.org
> […](#)
> On Mar 18, 2020, at 1:02 PM, édouard ***@***.***> wrote: Blockquotes are unstyled; just wondering if you have an opinion on that ... I prefer it to excessive margins, but maybe we want to italicise them or provide some sort of style hint that it's a quote? This was how we had previous specified quotes to look: Is this something that'd be implemented via main.scss? — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub, or unsubscribe.

Is there an additional build step required for having changes in css propagate to the markdown'd pages? I've edited the right location (I believe), but I'm not seeing the styling change.

matildepark commented:
Sometimes you gotta cancel and reserve the Zola build …

—
~haddef-sigwen
https://urbit.org

> On Mar 18, 2020, at 1:11 PM, édouard <notifications@github.com> wrote:
> 
> 
> Yeah that’s easy. border-left, padding-left, font-style: italic, on I think body.essay’s blockquotes — probably already in scss — ~haddef-sigwen https://urbit.org
> …
> On Mar 18, 2020, at 1:02 PM, édouard @.***> wrote: Blockquotes are unstyled; just wondering if you have an opinion on that ... I prefer it to excessive margins, but maybe we want to italicise them or provide some sort of style hint that it's a quote? This was how we had previous specified quotes to look: Is this something that'd be implemented via main.scss? — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> Is there an additional build step required for having changes in css propagate to the markdown'd pages? I've edited the right location (I believe), but I'm not seeing the styling change.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

urcades committed:
posts/precepts, main.scss: edited some quote formatting, added new styling for blockquotes

Added a new selector in main.scss for the <blockquote> element, used in the latest "urbit precepts" post. Fixed some of the resulting formatting changes in the posts themselves.

matildepark commented:
Looks good.

urcades committed:
main.scss: removed bottom padding from essay/post <blockquote> elements

See title

urcades commented:
Did a small fix to the bottom margin of post blockquotes, but just waiting on checks to complete. Will turn to Asana to ask for when to post. cc @crrncllns

philipcmonk commented:
I do like the images!

urcades commented:
There are some deep cuts in here from AZ, hope it is appreciated – Sedona
is a lovely place to source landscape imagery from.

—
~fabled-faster
https://urbit.org/


On Wed, Mar 18, 2020 at 2:36 PM Philip Monk <notifications@github.com>
wrote:

> I do like the images!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/urbit.org/pull/376#issuecomment-600795578>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJD2Y47NTM2CYGZHLRZJW3RIEIDNANCNFSM4LOR7LMA>
> .
>
